### DISCUSSION TITLE
.NET DEB packages availability on Ubuntu

### DISCUSSION THREAD

**Discussion by gregolsky** [dotnet/runtime#103622] [Q&A]
Hello,

We are trying to use latest version of .NET whenever possible and update to latest versions as they show up. However we have an issue with official Ubuntu repositories not providing the latest version of .NET packages.

```
C:\work> docker run -it ubuntu:22.04 bash -c 'apt update -qq && apt-cache policy dotnet-runtime-8.0'
2 packages can be upgraded. Run 'apt list --upgradable' to see them.
dotnet-runtime-8.0:
  Installed: (none)
  Candidate: 8.0.5-0ubuntu1~22.04.1
  Version table:
     8.0.5-0ubuntu1~22.04.1 500
        500 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages
```

Even the Microsoft official image does not have 8.0.6:
```
# docker run -it mcr.microsoft.com/dotnet/runtime:8.0-jammy bash -c 'apt update -qq && apt-cache policy dotnet-runtime-8.0'
Unable to find image 'mcr.microsoft.com/dotnet/runtime:8.0-jammy' locally
8.0-jammy: Pulling from dotnet/runtime
7646c8da3324: Already exists
5ed28ba5b486: Already exists
cf576d754b1e: Already exists
870f702105da: Pull complete
bdc98dd31166: Pull complete
Digest: sha256:6f69b363a22526a66dba13ce6d2f33ca94fd9dbf2e0a77e71523243d813cb534
Status: Downloaded newer image for mcr.microsoft.com/dotnet/runtime:8.0-jammy
All packages are up to date.
dotnet-runtime-8.0:
  Installed: (none)
  Candidate: 8.0.5-0ubuntu1~22.04.1
  Version table:
     8.0.5-0ubuntu1~22.04.1 500
        500 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages
```

Is this normal? I suspect it may take some time for the package to reach official Ubuntu repos. Do you know how long would it take exactly?

I know there's `packages.microsoft.com`, but it brings even more confusion as there's 8.0.7 version available there!?
Followed https://learn.microsoft.com/en-us/linux/packages#debian-based-linux-distributions
![image](https://github.com/dotnet/runtime/assets/1541805/7dfe3ac2-b166-49d0-9c42-5a77ba47db16)

To summarize since .NET packages [are not going to be published at packages.microsoft.com feed for new Ubuntu versions](https://github.com/dotnet/core/discussions/9258) dotnet/core#9258 and Ubuntu Archive is slow to publish updates, where can we obtain latest packages from now on?

Please advise

**Comment by huoyaoyuan** [comment]
It's interesting that the version bundled with Visual Studio is updated to 8.0.6 today, a week after the release.

**Comment by jkotas** [comment]
@richlander

**Comment by asvishnyakov** [comment]
Looks fixed now after #104145, please check again
